src/bootstrap/test.rs

```diff
@@ -1387,6 +1387,7 @@ note: if you're sure you want to do this, please open an issue as to why. In the
         let mut flags = if is_rustdoc { Vec::new() } else { vec!["-Crpath".to_string()] };
         flags.push(format!("-Cdebuginfo={}", builder.config.rust_debuginfo_level_tests));
         flags.push(builder.config.cmd.rustc_args().join(" "));
+        flags = flags.into_iter().filter(|x| !x.is_empty()).collect();
 
         if let Some(linker) = builder.linker(target) {
             cmd.arg("--linker").arg(linker);
```

Comment:
I see some cases of `        cmd.arg("--host-rustcflags").arg(hostflags.join(" "));` below (and for targetflags) -- those also seem like they ought to be passing something closer to:

```
for flag in hostflags {
  cmd.arg("--host-rustcflags").arg(flag);
}
```